File: Data Structures & Algorithms/is-anagram/submission-0.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        ss, tt = {}, {}
        for let in s:
            if let in ss: ss[let] += 1
            else: ss[let] = 1
        for let in t:
            if let in tt: tt[let] += 1 
            else: tt[let] = 1
        if ss == tt: return True
        return False
        

File: Data Structures & Algorithms/is-palindrome/submission-0.cpp
#include <cctype>
class Solution {
public:
    bool isNotAlphanumeric(char letter) {
        if (('A' <= letter) and (letter <= 'Z')) return false;
        if (('a' <= letter) and (letter <= 'z')) return false;
        if (('0' <= letter) and (letter <= '9')) return false;
        return true;
    }

    bool isPalindrome(string s) {
        bool ans = true;
        int left = 0; int right = s.size() - 1;
        while (left < right){
            if (isNotAlphanumeric(s[left])) {
                left++;
                continue;
            }
            if (isNotAlphanumeric(s[right])) {
                right --;
                continue;
            }
            if (tolower(s[left]) != tolower(s[right])) ans = false;
            left++; right--;
        }
        return ans;
    }
};
